File: app/src/androidTest/java/com/lambdaschool/congressdata/DetailsActivityTest.java
package com.lambdaschool.congressdata;

import androidx.test.espresso.ViewAction;
import androidx.test.espresso.contrib.RecyclerViewActions;
import androidx.test.espresso.intent.Intents;
import androidx.test.rule.ActivityTestRule;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.core.AllOf.allOf;
import static org.junit.Assert.*;

public class DetailsActivityTest {
    @Rule
    public ActivityTestRule<MainActivity> mainActivityTestRule = new ActivityTestRule<>(MainActivity.class, false, true);

    @Before
    public void setUp() throws Exception {

        Intents.init();
    //    onView( allOf( withId( R.id.text_list_name ), withText( "Ralph Abraham" ) ) ).perform( click() );//this one works
   //     onView(withId( R.id.layout_list )  ).perform( RecyclerViewActions.scrollTo(hasDescendant (withText( "James Baird" )) ) );
     //   onView(withId( R.id.layout_list )  ).perform( RecyclerViewActions.actionOnItemAtPosition( 0,click() ) );
        onView(withId(R.id.card_view)).check(matches(hasDescendant(withText("James Baird"))));
    }

    @Test
    public void testFacebook() {
        onView(withId( R.id.profile_facebook )).perform( click() );

    }

    @Test
    public void testTwitter() {
        onView(withId( R.id.profile_twitter )).perform( click() );

    }


    @Test
    public void testOffice() {
        onView(withId( R.id.profile_map )).perform( click() );

    }


    @Test
    public void testName() {
        onView(withId( R.id.profile_name )).perform( click() );

    }


    @After
    public void onClose() {

        Intents.release();
    }
}

File: app/src/androidTest/java/com/lambdaschool/congressdata/DetailsActivityTest.kt
package com.lambdaschool.congressdata


import android.content.Intent
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import androidx.test.rule.ActivityTestRule
import org.junit.FixMethodOrder
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.MethodSorters


/**

 * id: A000374
 * firstName: Ralph
 * middleName: null
 * lastName: Abraham
 * district: 5
 * facebookAccount: CongressmanRalphAbraham
 * location: LA - District 5
 * missedVotesPct: 36.73
 * office: 417 Cannon House Office Building
 * party: Republican
 * phone: 202-225-4306
 * state: LA
 * twitterAccount: RepAbraham
 * votesWithPartyPct: 97.56
 * subcommittees: 8
 * Conservation and Forestry
 * General Farm Commodities and Risk Management
 * Military Personnel
 * Intelligence and Emerging Threats and Capabilities
 * Seapower and Projection Forces
 * Research and Technology
 * Space and Aeronautics
 * Investigations and Oversight
 */
@RunWith(AndroidJUnit4::class)
@LargeTest
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
class DetailsActivityTest {

    @Rule
    @JvmField
    var detailsActivityTestRule: ActivityTestRule<DetailsActivity> = object : ActivityTestRule<DetailsActivity>(DetailsActivity::class.java) {
        override fun getActivityIntent(): Intent {
            val intent = Intent()
            intent.putExtra("id", "A000374")
            return intent
        }

    }


//  @Test
//  fun profileNameTextView_ProfileNameIsDisplayName() {

//      onView(withId(R.id.profile_name)).check(matches(withText("Ralph Abraham")))

//  }


    @Test
    fun profilePartyTextView_CorrectParty() {

        onView(withId(R.id.profile_party)).check(matches(withText("Republican")))

    }
    @Test
    fun profilePhoneTextView_CorrectParty() {

        onView(withId(R.id.profile_phone)).check(matches(withText("202-225-4306")))

    }


 // @Test
 // fun profileDistrictTextView_DistrictIsStateAndDistrict() {

 //     onView(withId(R.id.profile_district)).check(matches(withText("LA - District 5")))

 // }


//    @Test
//    fun profilePhoneTextView_Success() {
//
//        onView(withId(R.id.profile_phone)).check(matches(withText("202-225-8490")))
//
//    }

}